packages/opencensus-web-instrumentation-zone/test/test-interaction-tracker.ts

```diff
@@ -0,0 +1,408 @@
+    const onclick = () => {
+      expect(incrementTaskCountSpy.calls.count()).toBe(1);
+      expect(decrementTaskCountSpy.calls.count()).toBe(0);
+      const promise = getPromise();
+      promise.then(() => {
+        expect(incrementTaskCountSpy.calls.count()).toBe(2);
+        expect(decrementTaskCountSpy.calls.count()).toBe(1);
+      });
+    };
+    fakeInteraction(onclick);
+
+    onEndSpanSpy.and.callFake((rootSpan: Span) => {
+      expect(incrementTaskCountSpy.calls.count()).toBe(2);
+      expect(decrementTaskCountSpy.calls.count()).toBe(2);
+      expect(rootSpan.name).toBe('test interaction');
+      expect(rootSpan.ended).toBeTruthy();
+      expect(rootSpan.duration).toBeLessThanOrEqual(5);
+      done();
+    });
+  });
+
+  it('should handle interactions with canceled tasks', done => {
+    const canceledTask = () => {
+      const timeoutId = setTimeout(noop, 10);
+      expect(incrementTaskCountSpy.calls.count()).toBe(2);
+      expect(decrementTaskCountSpy.calls.count()).toBe(0);
+      setTimeout(() => {
+        expect(incrementTaskCountSpy.calls.count()).toBe(3);
+        expect(decrementTaskCountSpy.calls.count()).toBe(1);
+        // Cancel task before it finishes
+        clearTimeout(timeoutId);
+        expect(decrementTaskCountSpy.calls.count()).toBe(2);
+      }, 6);
+    };
+    fakeInteraction(canceledTask);
+
+    onEndSpanSpy.and.callFake((rootSpan: Span) => {
+      expect(incrementTaskCountSpy.calls.count()).toBe(3);
+      expect(decrementTaskCountSpy.calls.count()).toBe(3);
+      expect(rootSpan.name).toBe('test interaction');
+      expect(rootSpan.ended).toBeTruthy();
+      expect(rootSpan.duration).toBeLessThanOrEqual(12);
+      done();
+    });
+  });
+
+  it('should ignore interactions on elements with disable', done => {
+    const onclick = () => {
+      // No change in task count as the click is not tracked.
+      expect(incrementTaskCountSpy.calls.count()).toBe(0);
+      expect(decrementTaskCountSpy.calls.count()).toBe(0);
+
+      setTimeout(() => {
+        // No change in task count as the click is not tracked.
+        expect(incrementTaskCountSpy.calls.count()).toBe(0);
+        expect(decrementTaskCountSpy.calls.count()).toBe(0);
+      });
+    };
+    const button = createButton('test interaction', true);
+    fakeInteraction(onclick, button);
+
+    const onclick2 = () => {
+      // Click is detected as this button does is not 'disabled'.
+      expect(incrementTaskCountSpy.calls.count()).toBe(1);
+      expect(decrementTaskCountSpy.calls.count()).toBe(0);
+      setTimeout(() => {
+        expect(incrementTaskCountSpy.calls.count()).toBe(2);
+        expect(decrementTaskCountSpy.calls.count()).toBe(1);
+      });
+    };
+
+    button.removeAttribute('disabled');
+    fakeInteraction(onclick2, button);
+    onEndSpanSpy.and.callFake((rootSpan: Span) => {
+      // There should be only 2 counted tasks related to the second interaction.
```

Comment:
Could you also test this by making sure that `onEndSpanSpy` is only called once?